src/compile/mark/valueref.ts

```diff
@@ -83,18 +75,22 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
+      if (scale.type === 'bin-linear') {
+        return {
+          signal: `(scale("${scaleName}", ${field(channelDef, {binSuffix: 'start', datum: true})}) + scale("${scaleName}", ${field(channelDef, {binSuffix: 'end', datum: true})}))/2`
```

Comment:
I still feel like the old binMidSignal make it easy to read without the need for comment -- not sure why you wanna remove it.